bundle/config/mutator/resolve_variable_references.go

```diff
@@ -189,3 +203,28 @@ func (m *resolveVariableReferences) Apply(ctx context.Context, b *bundle.Bundle)
 	}
 	return diags
 }
+
+type cleanup func() error
+
+func updateWorkspacePathForSourceLinkedDeployment(b *bundle.Bundle) (cleanup, error) {
+	originalWorkspacePath := dyn.NilValue
+	fieldName := "workspace.file_path"
+
+	// If source-linked deployment is enabled, update the workspace path to the sync root path so variables will have correct path.
+	err := b.Config.Mutate(func(v dyn.Value) (dyn.Value, error) {
```

Comment:
Instead of mutating config here and reverting it back I suggest to change `lookup` function above to check if we look up for `workspace.file_path` and if we in `SourceLinkedDeployment` mode and just return a different value. Then the change becomes a read only with no changes to config itself

Reply:
> I don't like that we rely on dynamic form of config.presets.source_linked_deployment otherwise

We do rely on it in any case, right? We need to know if it's enable or not

Passing the reference to bundle is fine.

@pietern @denik WDYT about the approach here?